### PR TITLE
Add Navigator to JSON View

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
With really large event histories (tens of thousands), the JSON view will hang and crash the browser since we show all events and use syntax highlighting.

This PR prevents that from happening by rendering only one event at a time but with controls to navigate quickly across all the events. The controls will be:

1. Left/right arrow keys
2. Left/right arrows to click
3. Range Input (slider)
4. Input to pick a specific event id

### Screenshots (if applicable) 📸 

https://github.com/temporalio/ui/assets/7967403/fed97bf3-73bc-4890-b5e7-2a3a8a53df1f

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### CODE REVIEW

**Comment by rossedfort** [comment]
Lgtm, only thing I would say is that maybe `workflow-json-navigator.svelte` should go in `src/lib/components` instead of `src/lib/holocene` cause it has some business logic

**Review by rossedfort — APPROVED** [review]

File: src/lib/components/workflow/workflow-json-navigator.svelte [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        on:click={() => {
+          index += 1;
+        }}
+        aria-label="next"
+      >
+        <span
+          class="arrow arrow-right"
+          class:arrow-right-disabled={index === events.length}
+        />
+      </button>
```
Comment (laurakwhit) [inline src/lib/components/workflow/workflow-json-navigator.svelte:71]: Adding another `div` with breakpoints here makes it a bit easier to view on smaller screens 

Screenshot 2023-05-12 at 11 21 19 AM

```suggestion
  <div class="flex gap-4 max-sm:flex-col">
    <div class="flex w-full gap-4">
      <RangeInput
        id="range-input-event-history"
        min={1}
        max={events.length}
        bind:value={index}
      />
      <div class="flex items-center justify-center gap-3">
        <button
          class="caret"
          disabled={index === 1}
          on:click={() => {
            index -= 1;
          }}
          aria-label="previous"
        >
          <span
            class="arrow arrow-left"
            class:arrow-left-disabled={index === 1}
          />
        </button>
        <button
          class="caret"
          disabled={index === events.length}
          on:click={() => {
            index += 1;
          }}
          aria-label="next"
        >
          <span
            class="arrow arrow-right"
            class:arrow-right-disabled={index === events.length}
          />
        </button>
      </div>
```

File: src/lib/pages/workflow-history-json.svelte [hunk 2]
```diff
@@ -1,47 +1,46 @@
...
-      />
-    </label>
-  </div>
-  <CodeBlock content={events} data-testid="event-history-json" />
-{/await}
+<WorkflowJsonNavigator {events}>
+  <label
+    slot="decode"
+    for="decode-event-history"
+    class="flex items-center gap-4 font-secondary text-sm w-96"
```
Comment (laurakwhit) [inline src/lib/pages/workflow-history-json.svelte:39]: Just some small improvements for responsiveness
```suggestion
    class="flex items-center gap-4 font-secondary text-sm w-96 min-w-fit sm:justify-end"
```

File: src/lib/pages/workflow-history-json.svelte [hunk 3]
```diff
@@ -1,47 +1,46 @@
...
+    ><ToggleSwitch
+      id="decode-event-history"
+      bind:checked={decodeEventHistory}
+      data-testid="decode-event-history-toggle"
+    />View decoded event history
```
Comment (laurakwhit) [inline src/lib/pages/workflow-history-json.svelte:44]: Some wonkiness ensues when the workflow is still `Running` and events are being added. If you slide the slider and then toggle `View decoded event history` the number of events changes but the position of the number above the slider doesn't ⬇️ 

Screenshot 2023-05-12 at 11 42 39 AM

**Review by laurakwhit — APPROVED** [review]
Nice use of the `Slider` 💪 Some small UI things below, but other than that lgtm!

File: src/lib/components/workflow/workflow-json-navigator.svelte (hunk 1 above)
Comment (laurakwhit) [reply src/lib/components/workflow/workflow-json-navigator.svelte:71]: Updated in https://github.com/temporalio/ui/pull/1361

File: src/lib/pages/workflow-history-json.svelte (hunk 2 above)
Comment (laurakwhit) [reply src/lib/pages/workflow-history-json.svelte:39]: Updated in https://github.com/temporalio/ui/pull/1361

File: src/lib/pages/workflow-history-json.svelte (hunk 3 above)
Comment (laurakwhit) [reply src/lib/pages/workflow-history-json.svelte:44]: Fixed in https://github.com/temporalio/ui/pull/1361

**Pull request merged**